SignalMessaging/attachments/AttachmentApprovalViewController.swift

```diff
@@ -0,0 +1,171 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+import MediaPlayer
+
+@objc
+public protocol AttachmentApprovalViewControllerDelegate: class {
```

Comment:
The old version of AttachmentApprovalViewController took a completion handler, but I went for the delegate approach instead so we could have more flexibility.